Fix #144. Add support for Tempfile

javierav opened the pull request:
If we want to use this gem with an SVG stored in Active Storage, we have to temporarily download the file and then use it with gem helper

```ruby
module SvgHelper
  def remote_svg(blob, **)
    blob.open do |file|
      inline_svg_tag file, **
    end
  end
end

# in views
<%= remote_svg(user.icon) %>
```

The problem is that Active Storage uses a temporary file with the `Tempfile` class that uses `DelegateClass` over `File`, so technically it behaves like a `File` but does not inherit from it, so it could not be used with this gem.

This PR fixes that issue by providing support for `Tempfile`.

tagliala commented on lib/inline_svg/io_resource.rb:
```diff
@@ -3,14 +3,7 @@
-      when IO then 1
-      end
+      object.is_a?(IO) || object.is_a?(StringIO) || object.is_a?(Tempfile)
     end
```
hello, I understand that `default_for` appears not to be used by this library and it is not even tested, but is there a reason to remove it entirely?

javierav replied:
Great, I think it's a great option. As soon as the other PR is up, I'll update this one!

tagliala commented:
@javierav can you please rebase and add an entry in the changelog?

javierav commented:
Done @tagliala! I've simplified the commit and PR name to refer only `Tempfile`. I've also added a simple entry to the readme on how to use it with ActiveStorage. Feel free to add a better description if you see fit. We look forward to hearing from @jamesmartin!

jamesmartin reviewed (approved):
Thanks for the PR, @javierav. Agree with @tagliala that this change is worth merging as-is. 👍
